common/src/main/kotlin/io/homeassistant/companion/android/common/data/network/NetworkHelper.kt

```diff
@@ -0,0 +1,42 @@
+package io.homeassistant.companion.android.common.data.network
+
+import android.net.NetworkCapabilities
+import android.os.Build
+import javax.inject.Inject
+import javax.inject.Singleton
+
+interface NetworkHelper {
+    /** Returns if network is validated for internet access */
+    fun isNetworkValidated(): Boolean
+
+    /** Returns if any network connection exists */
+    fun hasActiveNetwork(): Boolean
+
+    /** Returns if the active data connection is using ethernet */
+    fun isUsingEthernet(): Boolean
+
+    /** Returns if the active data connection is a VPN connection */
+    fun isUsingVpn(): Boolean
+}
+
+@Singleton
+class NetworkHelperImpl @Inject constructor(private val capabilitiesChecker: NetworkCapabilitiesChecker) :
+    NetworkHelper {
```

Comment:
```suggestion
@Singleton
private class NetworkHelperImpl @Inject constructor(private val capabilitiesChecker: NetworkCapabilitiesChecker) :
    NetworkHelper {
```

Any reason to have a Singleton?